### PR TITLE
feat: Add vectorsearch entities

### LINKED ISSUE #1978: (langchain) Create vectorstore entities
As with https://github.com/newrelic/node-newrelic/issues/1963, we need entities for representing vectorstore searches and search results:

```js
class LlmVectorSearch {
  span_id
  trace_id
  transaction_id
  id = crypto.randomUUID()
  vendor = "langchain"
  vendor_source = "Node"
  ['request.query']
  ['request.k']
  duration
  ['response.number_of_documents']
  appName
}
```

```js
class LlmVectorSearchResult {
  span_id
  trace_id
  transaction_id
  id = crypto.randomUUID()
  vendor = "langchain"
  vendor_source = "Node"
  search_id = crypto.randomUUID()
  sequence = 0 // 1, 2, ... n
  page_content
  ['metadata.<key>']
  appName
}
```

See [the CDD](https://newrelic.atlassian.net/wiki/spaces/INST/pages/3162603539/Node.js+Agent+LangChain+Instrumentation) for further details.

### PR DESCRIPTION
**Pull request by svetlanabrennan**
## Description

Based this pr on the message entities pr: https://github.com/newrelic/node-newrelic/pull/1983

## How to Test

Added unit tests. 

## Related Issues

Closes #1978

### CODE REVIEW

File: lib/llm-events/langchain/vector-search-result.js
Comment (jsumners-nr) [inline lib/llm-events/langchain/vector-search-result.js:20]:
```suggestion
 */
const defaultParams = {
```

File: lib/llm-events/langchain/vector-search-result.js
Comment (jsumners-nr) [inline lib/llm-events/langchain/vector-search-result.js:23]: The `const crypto = require('crypto')` line is missing.

File: lib/llm-events/langchain/vector-search-result.js
Comment (jsumners-nr) [inline lib/llm-events/langchain/vector-search-result.js:35]: I don't think we need to make `search_id` a parameter. I believe it is an internal only field, so we can simplify things by:

```suggestion
const defaultParams = {
  pageContent: '',
  sequence: 0
}

class LangChainVectorSearchResult extends LangChainEvent {
  search_id = crypto.randomUUID()
  
  constructor(params) {
    params = Object.assign({}, defaultParams, params)
    super(params)

    this.page_content = params.pageContent
    this.sequence = params.sequence
  }
}
```

**Review by jsumners-nr — COMMENTED** [review]
Looking good. I have only a couple of suggestions.

File: lib/llm-events/langchain/vector-search-result.js
Comment (svetlanabrennan) [reply lib/llm-events/langchain/vector-search-result.js:34]: Sounds good.

File: lib/llm-events/langchain/vector-search-result.js
Comment (svetlanabrennan) [reply lib/llm-events/langchain/vector-search-result.js]: Ah yes. Fixed.

**Review by jsumners-nr — CHANGES_REQUESTED** [review]
Putting a block on this until #1983 is merged (this one depends on that one).

**Review by jsumners-nr — APPROVED** [review]
Looks good to me.

**Pull request merged**
